lib/src/client.dart

```diff
@@ -537,6 +539,37 @@ class Client {
     PaginationParams paginationParams = const PaginationParams(limit: 10),
     int messageLimit,
     bool onlyOffline = false,
+  }) {
+    final hash = base64.encode(utf8.encode(
+        '$filter${sort?.map((s) => s.toJson())}$options${paginationParams?.toJson()}$messageLimit$onlyOffline'));
+
+    if (_queryChannelsStreams.containsKey(hash)) {
+      return _queryChannelsStreams[hash];
+    }
+
+    final s = _doQueryChannels(
+      filter: filter,
+      sort: sort,
+      options: options,
+      paginationParams: paginationParams,
+      messageLimit: messageLimit,
+      onlyOffline: onlyOffline,
+    ).doOnDone(() {
+      return _queryChannelsStreams.remove(hash);
+    });
+
+    _queryChannelsStreams[hash] = s;
+
+    return s;
+  }
+
+  Stream<List<Channel>> _doQueryChannels({
```

Comment:
Consider marking these parameters with the `@required` annotation. Since they are all used by the function, we can take advantage of the dart analyzer to help us anytime we use it.